projects/js-packages/publicize-components/src/components/social-image-generator/template-picker/index.js

```diff
@@ -1,80 +1,58 @@
-import { ThemeProvider } from '@automattic/jetpack-components';
-import { Button, Modal } from '@wordpress/components';
-import { useState, useCallback, useMemo } from '@wordpress/element';
+import { useState, useMemo } from '@wordpress/element';
 import { sprintf, __ } from '@wordpress/i18n';
 import classnames from 'classnames';
 import React from 'react';
 import styles from './styles.module.scss';
 import TEMPLATES_DATA from './templates.js';
 
 /**
- * A component that displays a modal with a list of templates to choose from.
  *
- * @param {object} props - The component props.
- * @param {Function} props.onSelect - A function that will be called when a template is selected. Receives the name of the selected template as an argument.
- * @param {Function} props.render - A function that will be called with an object containing an "open" function, which can be called to open the template picker.
- * @param {string|null} [props.value=null] - The name of the currently selected template.
+ * The pure template picker component. Does not save the template changes, just sends it back to the parent component,
+ * with the onTemplateSelected callback.
+ *
+ * @param {{value: string|null, onTemplateSelected: Function}} props - The component props:
+ * Value is the name of the currently selected template, onTemplateSelected is a function that
+ * will be called when a template is selected. Receives the name of the selected template as an argument.
  * @returns {React.ReactNode} - The component's rendered output.
  */
-const TemplatePicker = ( { onSelect, render, value = null } ) => {
-	const [ isOpen, setIsOpen ] = useState( false );
+const TemplatePicker = ( { value = null, onTemplateSelected = null } ) => {
 	const [ selectedTemplate, setSelectedTemplate ] = useState( value );
 
-	const openPicker = useCallback( () => setIsOpen( true ), [ setIsOpen ] );
-	const closePicker = useCallback( () => {
-		setSelectedTemplate( value );
-		setIsOpen( false );
-	}, [ value, setSelectedTemplate, setIsOpen ] );
-	const saveAndClosePicker = useCallback( () => {
-		onSelect( selectedTemplate );
-		setIsOpen( false );
-	}, [ onSelect, setIsOpen, selectedTemplate ] );
-
 	// Add memoized callback function for each template.
 	const TEMPLATES = useMemo(
 		() =>
 			TEMPLATES_DATA.map( template => ( {
 				...template,
-				onSelect: () => setSelectedTemplate( template.name ),
+				onSelect: () => {
+					setSelectedTemplate( template.name );
+					if ( onTemplateSelected ) {
```

Comment:
I wonder if there is a more succinct way to do this, especially if we remove the internal state.

Perhaps we could do something like:

```js
onSelect: () => onSelectTemplate?.( template.name )
```

I don't know if that's better or not!